modules/salesforce_encrypt/src/Rest/RestClient.php

```diff
@@ -15,258 +15,15 @@
 use Drupal\encrypt\EncryptionProfileManagerInterface;
 use Drupal\salesforce\EntityNotFoundException;
 use Drupal\salesforce\Rest\RestClient as SalesforceRestClient;
+use Drupal\salesforce\SalesforceAuthProviderPluginManager;
 use GuzzleHttp\ClientInterface;
 use Drupal\Component\Datetime\TimeInterface;
 
 /**
  * Objects, properties, and methods to communicate with the Salesforce REST API.
+ *
+ * @deprecated use \Drupal\salesforce\SalesforceAuthProviderPluginManager::getConfig() to access the current active auth configuration.
```

Comment:
I would rather have a breaking change here, because nobody should be extending or using this class any more, and remove it altogether in 3.2.

I've updated the commit to keep the interface in place, and throw Deprecation exceptions when any of these methods are called.